templates/typescript_gapic/src/$version/$service_client.ts.njk

```diff
@@ -352,12 +352,24 @@ export class {{ service.name }}Client {
         protosTypes{{ method.inputInterface }}|undefined, {}|undefined
       ]>|void {
     request = request || {};
-    let options = optionsOrCallback;
-    if (typeof options === 'function' && callback === undefined) {
-      callback = options;
+    let options: gax.CallOptions;
+    if (typeof optionsOrCallback === 'function' && callback === undefined) {
+      callback = optionsOrCallback;
       options = {};
     }
+    else {
+      options = optionsOrCallback as gax.CallOptions;
+    }
     options = options || {};
+{%- if method.headerRequestParams.length > 0 %}
+    options.otherArgs = options.otherArgs || {};
+    options.otherArgs.headers = options.otherArgs.headers || {};
+    options.otherArgs.headers[
+      'x-goog-request-params'
+    ] = gax.routingHeader.fromParams({
+      '{{ method.headerRequestParams.toString().toCamelCase() }}': request.{{ method.headerRequestParams.camelCaseBeforeDot() }} || '',
```

Comment:
It should be snake_case for the left part and camelCase for the right part. Left part is basically taken from proto as is (since it gets passed to gRPC server as a header), right part is converted to camelCase because it's how `protobufjs` processes keys with underscore.

Example here: https://github.com/googleapis/nodejs-logging/blob/dcb48172656de8b449a2c05e8cbe999ebaac90bf/src/v2/metrics_service_v2_client.js#L460

Reply:
Actually, if we discuss it, it would be nice to have it covered by a baseline test :)

Reply:
sounds great! but do you mean unit test? we have` keymanager ` `translate`, `text-to-speech`API which respects this routing header, do we want to add one more baseline test, or just test this functionality in the `unit/`? Thanks!

Reply:
Maybe add `logging`? (in a separate PR)